src/cache/file_cache.h

```diff
@@ -101,24 +102,55 @@ class FileCache {
                                   string::CommandLine command_line,
                                   string::Version version);
 
+  static string::HandledHash Hash(string::HandledSource code,
```

Comment:
Unit tests still use the methods without `extra_files`. I'm not sure if changing all the calls there is better than just having two overloads of these methods.